pkg/services/ngalert/api/fork_ruler.go

```diff
@@ -2,34 +2,64 @@ package api
 
 import (
 	"fmt"
+	"strconv"
 
 	apimodels "github.com/grafana/alerting-api/pkg/api"
 	"github.com/grafana/grafana/pkg/api/response"
 	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/datasources"
+)
+
+type backendType string
+
+const (
```

Comment:
I'm unclear on the need for new types here. Can't we just use the `Lotex` or `Grafana` backend types exposed by `apimodels`? I believe we only need to distinguish between grafana managed and datasource managed (lotex) rules.

Reply:
Returning backendType (string) instead of int it's only useful for passing the unsupported datasource type to the error response like here:
https://github.com/grafana/grafana/blob/57ae8851315ee7e5cd109c93ddde5f1960f5d737/pkg/services/ngalert/api/fork_ruler.go#L65

so, for instance, in the following example (assuming that datasource with id 1 is a graphite one) we get this response:
```
curl  admin:admin@localhost:3000/ruler/1/api/v1/rules/test
{"message":"unexpected backend type (graphite)"}
```

Reply:
on the other hand, maybe we wouldn't like to reveal such an information about the datasource. WDYT?

Reply:
Let's just implement the `String` method on the original type.